inc/ticket.class.php

```diff
@@ -2571,18 +2707,16 @@ function getSearchOptions() {
             $tokeep[] = 'validation';
          }
          $keep = false;
-         foreach ($tab as $key => $val) {
-            if (!is_array($val)) {
-               $keep = in_array($key, $tokeep);
+         foreach ($tab as $key => &$val) {
+            if (count($val) == 2) {
```

Comment:
if think testing the presence (or absence) of field/table key could be preferable